crypto/s2n_ecc_evp.c

```diff
@@ -157,11 +157,14 @@ static int s2n_ecc_evp_compute_shared_secret(EVP_PKEY *own_key, EVP_PKEY *peer_p
     POSIX_ENSURE_REF(peer_public);
     POSIX_ENSURE_REF(own_key);
 
-    /* From RFC 8446 Section 4.2.8.2: For the curves secp256r1 and secp384r1 peers MUST validate each other's
-     * public value Q by ensuring that the point is a valid point on the elliptic curve.
-     * For the curve x25519 the peer public-key validation check doesn't apply.
+    /* From RFC 8446(TLS1.3) Section 4.2.8.2: For the curves secp256r1, secp384r1, and secp521r1, peers MUST validate 
+     * each other's public value Q by ensuring that the point is a valid point on the elliptic curve.
+     * For the curve x25519 and x448 the peer public-key validation check doesn't apply.
+     * From RFC 8422(TLS1.2) Section 5.11: With the NIST curves, each party MUST validate the public key sent by its peer
```

Comment:
There are more non-NIST curves than X25519 and X448 (e.g. Koblitz curves like sect571k1). Not sure how we best model the != vs the == though. My vote would probably be to keep the != (less chance of accidentally forgetting to update this again) and just add another comment.

Reply:
IMO we should go with `!=` instead of `==` here to prevent any future curves from missing the check. 

I could add the following comment:

```
Note that the EC_KEY_check_key check is a MUST for only NIST curves, if a non-NIST curve is added to s2n-tls, 
the EC_KEY_check_key is an additional validation step that increases security but decreases performance. 
```